crate/utils/src/crypto/hybrid_encryption_system.rs

```diff
@@ -205,80 +234,71 @@ impl EncryptionSystem for EciesEncryption {
 
 /// Decrypt a single block of data encrypted using a Salsa Sealed Box
 /// Cannot be used as a stream decipher
-pub struct EciesDecryption {
+pub struct HybridDecryptionSystem {
     private_key_uid: String,
-    private_key_bytes: Vec<u8>,
-    recommended_curve: RecommendedCurve,
+    private_key: Object,
+    // recommended_curve: RecommendedCurve,
 }
 
-impl EciesDecryption {
+impl HybridDecryptionSystem {
     pub fn instantiate(
         private_key_uid: &str,
         private_key: &Object,
     ) -> Result<Self, KmipUtilsError> {
-        debug!("instantiate: entering");
-        let recommended_curve = private_key
-            .attributes()?
-            .cryptographic_domain_parameters
-            .ok_or(KmipUtilsError::NotSupported(
-                "Private key without cryptographic domain parameters is not supported".to_string(),
-            ))?
-            .recommended_curve
-            .ok_or(KmipUtilsError::NotSupported(
-                "Private key without recommended_curve is not supported".to_string(),
-            ))?;
-
-        debug!("Instantiated ECIES decipher for user decryption key id: {private_key_uid}");
+        debug!(
+            "Instantiated hybrid decryption system for user decryption key id: {private_key_uid}"
+        );
 
         Ok(Self {
             private_key_uid: private_key_uid.into(),
-            private_key_bytes: private_key.key_block()?.key_bytes()?,
-            recommended_curve,
+            private_key: private_key.clone(),
+            // recommended_curve,
         })
```

Comment:
Looking at these changes, it doesn't seem to be interesting to have this `instantiate` anymore, what do you think?
One could just init `HybridDecryptionSystem` just by filling the fields

Reply:
you are right, thank you